### PR TITLE
RSF negative energy fix

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
fix: It is no longer possible for Service borgs using the Rapid Service Fabricator to end up with negative energy.
/ 🆑

### CODE REVIEW

**Comment by The-Driker** [comment]
maximum burger

WHEN BURGERS GO TOO FAR!

File: code/game/objects/items/weapons/RSF.dm [hunk 1]
```diff
@@ -67,18 +67,24 @@ RSF
 		to_chat(user, "The RSF can only create service items on tables, or floors.")
 		return
 
-	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
-	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
-	var type_path = configured_items[mode][3]
-	new type_path(spawn_location)
 	if(isrobot(user))
 		var/mob/living/silicon/robot/engy = user
+		if (engy.cell.charge < configured_items[mode][2])
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RSF.dm]: `if (`

Stop this now.

File: code/game/objects/items/weapons/RSF.dm [hunk 2]
```diff
@@ -67,18 +67,24 @@ RSF
...
-	var type_path = configured_items[mode][3]
-	new type_path(spawn_location)
 	if(isrobot(user))
 		var/mob/living/silicon/robot/engy = user
+		if (engy.cell.charge < configured_items[mode][2])
+			to_chat(user, "<span class='warning'>Insufficient energy.</span>")
+			return
 		engy.cell.charge -= configured_items[mode][2]
 	else
+		if (matter == 0)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RSF.dm]: cease!

File: code/game/objects/items/weapons/RSF.dm [hunk 3]
```diff
@@ -67,18 +67,24 @@ RSF
...
+			to_chat(user, "<span class='warning'>Insufficient matter.</span>")
+			return
 		matter--
 		to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
 		desc = "A RSF. It currently holds [matter]/30 fabrication-units."
 
+	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
+	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+	var type_path = configured_items[mode][3]
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/RSF.dm]: what are you doinggggg

File: code/game/objects/items/weapons/RSF.dm (hunk 1 above)
Comment (Kyep) [reply code/game/objects/items/weapons/RSF.dm]: Changed

File: code/game/objects/items/weapons/RSF.dm (hunk 2 above)
Comment (Kyep) [reply code/game/objects/items/weapons/RSF.dm]: Changed

File: code/game/objects/items/weapons/RSF.dm (hunk 3 above)
Comment (Kyep) [reply code/game/objects/items/weapons/RSF.dm]: Hey, this one isn't mine. I just moved it. It was already that way!
But yeah, changed.

**Comment by Kyep** [comment]
@Fox-McCloud All three of your requested changes are done.

File: code/game/objects/items/weapons/RSF.dm [hunk 4]
```diff
@@ -67,18 +67,24 @@ RSF
...
 	else
+		if(matter == 0)
+			to_chat(user, "<span class='warning'>Insufficient matter.</span>")
+			return
 		matter--
 		to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
 		desc = "A RSF. It currently holds [matter]/30 fabrication-units."
 
+	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
+	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/RSF.dm]: No src. please (yeah I know you just moved the code)

File: code/game/objects/items/weapons/RSF.dm [hunk 5]
```diff
@@ -67,18 +67,24 @@ RSF
...
-	var type_path = configured_items[mode][3]
-	new type_path(spawn_location)
 	if(isrobot(user))
 		var/mob/living/silicon/robot/engy = user
+		if(engy.cell.charge < configured_items[mode][2])
+			to_chat(user, "<span class='warning'>Insufficient energy.</span>")
+			return
 		engy.cell.charge -= configured_items[mode][2]
 	else
+		if(matter == 0)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/RSF.dm]: if(!matter)

File: code/game/objects/items/weapons/RSF.dm [hunk 6]
```diff
@@ -67,18 +67,24 @@ RSF
 		to_chat(user, "The RSF can only create service items on tables, or floors.")
 		return
 
-	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
-	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
-	var type_path = configured_items[mode][3]
-	new type_path(spawn_location)
 	if(isrobot(user))
 		var/mob/living/silicon/robot/engy = user
+		if(engy.cell.charge < configured_items[mode][2])
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/RSF.dm]: Isn't the correct way to do this
```
if(!engy.cell.use(configured_items[mode][2]))
    to_chat(user, "<span class='warning'>Insufficient energy.</span>")
    return

**Comment by Kyep** [comment]
@tigercat2000 All of your requests are done.

**Pull request merged**